package/kedro_viz/integrations/kedro/sqlite_store.py

```diff
@@ -60,10 +85,109 @@ def to_json(self) -> str:
+        to your local project.
+        Note: All the database files are deleted after they are merged to the main session_store.db.
+        """
+        try:
+            # In theory we should be able to do this as a single operation:
+            # self._remote_fs.get(f"{self.remote_location}/*.db", str(Path(self.location).parent))
+            # but this does not seem to work correctly - maybe a bug in fsspec. So instead
+            # we do it in two steps.
+            remote_dbs = self._remote_fs.glob(f"{self.remote_location}/*.db")
+            self._remote_fs.get(remote_dbs, str(Path(self.location).parent))
+        except Exception as exc:
+            logger.exception(exc)
+
+    def _merge(self):
+        """Merges all the session store databases stored at the specified locations into the user's local session_store.db
+
+        Notes:
+        - This method uses multiple SQLAlchemy engines to connect to the user's session_store.db and to all the other downloaded dbs.
+        - It is assumed that all the databases share the same schema.
+        - In the version 1.0 - we only merge the runs table which contains all the experiments.
+        - The downloaded database files are deleted after it's runs are merged with the user's local session_store.db
+
+        Args:
+            database_location:  A list of local filepath in string format for all the databases
+
+        """
+        # Connect to the user's local session_store.db
+        engine, session_class = create_db_engine(self.location)
+        Base.metadata.create_all(bind=engine)
+        database = next(get_db(session_class))
+
+        all_runs_data = []
+        existing_run_ids = [id[0] for id in database.query(RunModel.id).all()]
+
+        temp_engine = None
+        # Iterate through each downloaded database
+        databases_location = set(Path(self.location).parent.glob("*.db")) - {
+            Path(self.location)
+        }
+
+        for db_loc in databases_location:
+            if temp_engine:
+                temp_engine.dispose()
+            # Open a connection to the downloaded database
+            temp_engine = create_engine(f"sqlite:///{db_loc}")
+            with temp_engine.connect() as database_conn:
+                db_metadata = MetaData()
+                db_metadata.reflect(bind=temp_engine)
+                # Merge data from the 'runs' table
+                for table_name, table_obj in db_metadata.tables.items():
+                    if table_name == "runs":
+                        data = database_conn.execute(table_obj.select()).fetchall()
+                        for row in data:
+                            if row.id not in existing_run_ids:
+                                existing_run_ids.append(row.id)
+                                all_runs_data.append((row._asdict()))
+            temp_engine.dispose()
+
+        database.bulk_insert_mappings(RunModel, all_runs_data)
```

Comment:
Apparently this is a legacy method so probably we shouldn't use it. https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session.bulk_insert_mappings